model-optimizer/mo/front/common/partial_infer/utils.py

```diff
@@ -113,4 +113,34 @@ def broadcast_shape(first_shape, second_shape):
         assert a_val == 1 or b_val == 1 or a_val == b_val, "Input shape do not broadcast"
         new_val = b_val if a_val == 1 else a_val
         new_shape[-i - 1] = new_val
-    return int64_array(new_shape)
\ No newline at end of file
+    return int64_array(new_shape)
+
+
+def get_shape_from_slice(input_shape: np.ndarray, slices: List) -> np.ndarray:
+    """
+    Calculate shape of a tensor after slicing without actually creating the resulting tensor.
+    Is introduced to prevent potentially large memory consumption.
+    """
+    output_shape = []
+    num_new_axes = np.count_nonzero(list(map(lambda x: x is np.newaxis, slices)))
+    num_ellipsis_inserts = len(input_shape) - len(slices) + num_new_axes + 1
+
+    in_idx = 0
+    for i, s in enumerate(slices):
+        if isinstance(s, slice):
+            output_shape.append(len(range(*s.indices(input_shape[in_idx]))))
+            in_idx += 1
+        elif s is np.newaxis:
+            output_shape.append(1)
+        elif isinstance(s, int):  # shrink_axis
+            in_idx += 1
+        elif s is Ellipsis:
```

Comment:
what object is Ellipsis here?

Reply:
Ellipsis is a python keyword for `...` And it's a singleton.